### ISSUE TITLE
Is there a way to sync treestyle tab between 2 computers?

### ISSUE THREAD

**Issue by justcosmic1** [piroor/treestyletab#2585]
Is there a way to sync treestyle tab between 2 computers?
I am using Firefox's inbuilt sync feature and tabs sync - but Tree style tabs don't seem to.
I am wondering if this is possible (maybe I am just missing something)?
I'm sure there must be quite a few people who need to do this - I did a search here but couldn't find anything.
Many thanks in advance for any help on this :)

**Comment by irvinm** [comment]
TST does support Firefox's built in sync.  You should haven't to do anything other than make sure both computers are using TST and that sync is enabled on both.

**Comment by justcosmic1** [comment]
Well I only just did a new installation on the 2nd computer and then synced which installed my extensions and synced my bookmarks and tabs but not TST. However maybe I just need to restart and try again tomorrow - I'll update you then - in the meantime I'll remain hopeful knowing that it should work in theory! Thanks for getting back so quickly :)

**Comment by piroor** [comment]
TST stores some of its configurations to the sync-able storage, so on this context TST supports sync.

On the other hand, both TST and Firefox don't provide ability to synchronize the state of the tab bar. Firefox on a different device always starts with blank tab bar (or last tabs on the device), and you just can show the list of tabs opened in other devices and choose individual tabs to import from the list.

If you want same tabs are shown in Firefox on multiple devices, you'll need to install something addon like [Tab Session Manager](https://addons.mozilla.org/firefox/addon/tab-session-manager/) (sorry I don't use it so I don't know whether such a feature is actually contained or not.)

**Comment by irvinm** [comment]
@piroor good point ... I missed the specifics of "tabs" being synced.  (Obviously I meant TST settings just being synced)  Sorry about that @justcosmic1.

**Comment by irvinm** [comment]
@piroor I think we can close this one.

**Comment by justcosmic1** [comment]
OK so it seems that there isn't currently a way to sync TST directly. But just to be sure...
@piroor are you saying it's just general settings of TST that are synced or are they settings that related to the open tabs ? 
If I did have Tab Session manager I presume it still wouldn't save the tree info that TST has ? 
Other way of looking at it would be if there's a way to backup TST locally and then import it on a 2nd computer...not so ideal, but maybe a work around? 
Just thought I would ask these few questions before we close the thread - also if anyne else comes here looking for answers it will be clearer.

**Comment by irvinm** [comment]
@justcosmic1 TSM does support keeping TSTs tree structure when saving and restoring a session. (I use it ... you do have to enable support for TST within TSM's options.)  You could absolutely export a session from TSM from one machine, import it to another, then restore that session.

**Comment by piroor** [comment]
@justcosmic1 I'm personally satisfied the default "tabs in other devices" feature of Firefox itself, so I'm not familiar how to use Firefox (and TST) with synchronized tab bar on multiple devices.

Currently TST supports to save tree structure to bookmarks (you can do it via the context menu command "Bookmark this Tree" on the TST sidebar), and you can restore tree with a context menu command "Open All as a Tree" on the created bookmark folder. I think it may help you to take complete tree between devices, if you synchronize bookmarks.

**Comment by irvinm** [comment]
@piroor I think there are sufficient workarounds for this to be able to close this one.

**Comment by ReanimationXP** [comment]
I disagree that anything currently in the plugin constitutes as a "workaround" for actively keeping tabs and tab groups especially synced between two devices. Vertitabs has just introduced this and I may be switching because of it.

**Comment by piroor** [comment]
After this issue was closed I experimentally tried to provide more effective "sync tabs across multiple devices including both PC and mobile" feature based on WebExtensions API.
https://github.com/piroor/send-tabs-to-device-simulator
But finally the project was suspended due to WebExtensions API restrictions: Firefox for Android does not implement `storage.sync` so TST on Android can't send/receive any data via the WebExtensions API.

I'm not well about extensions API of Chrome on Android but it looks an advantage of Chrome if it provides `storage.sync` on Android - it is a behind of Firefox.

* I'm not planning to port TST to Chrome: TST strongly depends on Firefox specific APIs, and I currently use Firefox as my main browser.
* Of course we were able to provide more complete "sync" feature across devices with something custom sync server, but I'm negative to make TST more service-connected software. I hope to keep TST stand-alone based only on WebExtensions API.
